pkg/controllers/dynakube/logmonitoring/logmonsettings/reconciler.go

```diff
@@ -0,0 +1,82 @@
+package logmonsettings
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube/logmonitoring"
+	dtclient "github.com/Dynatrace/dynatrace-operator/pkg/clients/dynatrace"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/timeprovider"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+type reconciler struct {
+	dk  *dynakube.DynaKube
+	dtc dtclient.Client
+
+	timeProvider *timeprovider.Provider
+}
+
+type ReconcilerBuilder func(dtc dtclient.Client, dk *dynakube.DynaKube) controllers.Reconciler
+
+var _ ReconcilerBuilder = NewReconciler
+
+func NewReconciler(dtc dtclient.Client, dk *dynakube.DynaKube) controllers.Reconciler {
+	return &reconciler{
+		dk:           dk,
+		dtc:          dtc,
+		timeProvider: timeprovider.New(),
+	}
+}
+
+func (r *reconciler) Reconcile(ctx context.Context) error {
+	if !conditions.IsOutdated(r.timeProvider, r.dk, conditionType) {
+		return nil
+	}
+
+	if !r.dk.LogMonitoring().IsEnabled() {
+		meta.RemoveStatusCondition(r.dk.Conditions(), conditionType)
+	}
```

Comment:
I think we should add `return nil`here. Operator shouldn't send LM queries to the DT server if LogMonitoring is Not enabled?

Reply:
I agree, you are right. https://github.com/Dynatrace/dynatrace-operator/pull/4053/commits/2320f89c12bb9bfa9a873d7c75f1fa45105f9707